frontend/src/pages/projects/notebook/useRouteForNotebook.ts

```diff
@@ -19,12 +21,17 @@ const useRouteForNotebook = (
         return;
       }
       if (notebookName && projectName) {
-        getRoute(notebookName, projectName)
+        // if not using service mesh fetch openshift route, otherwise get Istio Ingress Gateway route
+        const getRoutePromise = dashboardConfig.spec.dashboardConfig.disableServiceMesh
```

Comment:
I have the same exact question as the logic in the backend, we are mixing a solution of annotations and logic, is this covering all these scenarios:

* Enabling service mesh and trying to get the route of old notebooks without annotation
* Deploy new notebooks under service mesh, disabling the feature and getting the route

Reply:
Good point, I think we shouldn't be checking the flag only. We should also check if the ns have the mesh annotation set. If both are true we should change the route to istio ingress. Otherwise assume old logic.

On that note, however: this will only work for users who have access to query `istio-system` for it. We are looking at a better way. Perhaps setting `'opendatahub.io/link'` annotation on the namespace by our controller should be sufficient. WDYT @lucferbux?

Reply:
Perfect, I saw recent updates, let me know when the pr is ready to review again and I'll look at it, thank you very much.